waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -913,23 +913,54 @@ proc addAll*(wakuRlnRelay: WakuRLNRelay, list: seq[IDCommitment]): RlnRelayResul
       return err(memberAdded.error())
   return ok()
 
-# the types of inputs to this handler matches the MemberRegistered event/proc defined in the MembershipContract interface
-type RegistrationEventHandler  = proc(pubkey: Uint256, index: Uint256): void {.gcsafe, closure, raises: [Defect].}
+type GroupUpdateHandler* = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.gcsafe, raises: [Defect].}
 
-proc subscribeToMemberRegistrations(web3: Web3, contractAddress: Address, handler: RegistrationEventHandler, fromBlock: string = "0x0"): Future[Subscription] {.async, gcsafe} =
-  var contractObj = web3.contractSender(MembershipContract, contractAddress)
-  return await contractObj.subscribe(MemberRegistered, %*{"fromBlock": fromBlock, "address": contractAddress}) do(pubkey: Uint256, index: Uint256){.raises: [Defect], gcsafe.}:
+proc generateGroupUpdateHandler(rlnPeer: WakuRLNRelay): GroupUpdateHandler =
+  # assuming all the members arrive in order
+  # TODO: check the index and the pubkey depending on
+  # the group update operation
+  var handler: GroupUpdateHandler
+  handler = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.raises: [Defect].} =
+    var pk: IDCommitment
     try:
-      debug "onRegister", pubkey = pubkey, index = index
-      handler(pubkey, index)
-    except Exception as err:
-      # chronos still raises exceptions which inherit directly from Exception
-      error "Error handling new member registration: ", err=err.msg
-      doAssert false, err.msg
-  do (err: CatchableError):
-    error "Error from subscription: ", err=err.msg
+      pk = pubkey.toIDCommitment()
+    except:
+      return err("invalid pubkey")
+    let isSuccessful = rlnPeer.insertMember(pk)
```

Comment:
It would be good to check the index of the received pk is one plus the `rlnPeer.lastSeenMembershipIndex`.

Reply:
Addressed in [582a6c3](https://github.com/status-im/nwaku/pull/1296/commits/582a6c322025d6d125f97ad434311170bb454c26)